blank_rust_project/contract/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_bindgen::{env, near_bindgen};
+use std::collections::HashMap;
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+pub struct Welcome {
+    records: HashMap<String, String>,
+}
+
+#[near_bindgen]
+impl Welcome {
+    pub fn set_greeting(&mut self, message: String) {
+        let account_id = env::signer_account_id();
+        self.records.insert(account_id, message);
+    }
+
+    pub fn welcome(&self, account_id: String) -> Option<String> {
+        if self.records.get(&account_id) == None {
+            env::log(b"using default message");
```

Comment:
Please add capitalization and punctuation.

Reply:
done